### PR TITLE
[BE2] Added RumorState handler

### PR DESCRIPTION
**Pull request by DanielTavaresA**
Pending on #1889 

When receiving a RumorState query, we send back a response with the list of rumors that are missing from the sender

### CODE REVIEW

**Review by onsriahi14 — DISMISSED** [review]
Clear and clean code!

File: be2-scala/src/main/scala/ch/epfl/pop/decentralized/GossipManager.scala [hunk 1]
```diff
@@ -7,15 +7,15 @@ import akka.pattern.AskableActorRef
...
+import ch.epfl.pop.model.network.method.{GreetServer, Rumor, RumorState}
 import ch.epfl.pop.model.network.{JsonRpcRequest, JsonRpcResponse, MethodType}
 import ch.epfl.pop.model.objects.{Base64Data, Channel, PublicKey, RumorData}
 import ch.epfl.pop.pubsub.AskPatternConstants
 import ch.epfl.pop.pubsub.ClientActor.ClientAnswer
 import ch.epfl.pop.pubsub.graph.validators.RpcValidator
 import ch.epfl.pop.pubsub.graph.{ErrorCodes, GraphMessage, PipelineError}
 import ch.epfl.pop.storage.DbActor
-import ch.epfl.pop.storage.DbActor.{DbActorAck, DbActorReadRumor, DbActorReadRumorData}
+import ch.epfl.pop.storage.DbActor.{DbActorAck, DbActorGenerateRumorStateAns, DbActorReadRumor, DbActorReadRumorData}
```
Comment (K1li4nL) [inline be2-scala/src/main/scala/ch/epfl/pop/decentralized/GossipManager.scala:18]: Not sure these imports are necessary ?

File: be2-scala/src/test/scala/ch/epfl/pop/model/network/method/RumorSuite.scala [hunk 2]
```diff
@@ -43,19 +43,4 @@ class RumorSuite extends FunSuite with Matchers {
...
-      PublicKey(Base64Data.encode("3")) -> 3
-    )
-
-    val rumorState: RumorState = RumorState(state)
-
-    val encodedDecoded = RumorState.buildFromJson(rumorState.toJsonString)
-
-    encodedDecoded.state shouldBe state
-  }
-
```
Comment (K1li4nL) [inline be2-scala/src/test/scala/ch/epfl/pop/model/network/method/RumorSuite.scala:60]: Why delete that  ?

**Review by K1li4nL — CHANGES_REQUESTED** [review]
Went quickly over it but looks good to me overall, just a few nitpicks and should be good to merge

File: be2-scala/src/test/scala/ch/epfl/pop/model/network/method/RumorSuite.scala (hunk 2 above)
Comment (DanielTavaresA) [reply be2-scala/src/test/scala/ch/epfl/pop/model/network/method/RumorSuite.scala:60]: Moved to RumorStateSuite apparently

File: be2-scala/src/main/scala/ch/epfl/pop/decentralized/GossipManager.scala (hunk 1 above)
Comment (DanielTavaresA) [reply be2-scala/src/main/scala/ch/epfl/pop/decentralized/GossipManager.scala:18]: Probably there from an old version, will remove

**Review by K1li4nL — APPROVED** [review]
LGTM

**Pull request merged**